src/main/java/org/spongepowered/common/service/sql/SqlServiceImpl.java

```diff
@@ -266,9 +278,15 @@ public static ConnectionInfo fromUrl(@Nullable PluginContainer container, String
             if (container != null && derelativizer != null) {
                 serverDatabaseSpecifier = derelativizer.apply(container, serverDatabaseSpecifier);
             }
+            final boolean hasOptions = match.group(6) != null;
+            final Set<String> rawOptions = hasOptions ? Sets.newHashSet(match.group(7).split("&")) : Collections.EMPTY_SET;
+            if (protocol.equals("mysql")) {
```

Comment:
Rather than messing about with options, could you not just set the name of the driver class to the MySQL driver so that we explicity set it, rather than pull apart all the options and force the MariaDB driver to parse it first? It seems to me that you're doing all this work to get a specific JDBC driver, without specifying that specific driver. Why say "MariaDB driver, reject this protocol and send it off to someone else" when you can say "I know I want MySQL to handle this, this is the class it needs to be"? 

I'd think all you need to do remove all this option parsing stuff and set the driverClass explicitly (also extracting the Driver class name into its own constant):

```java
private final static String MYSQL_DRIVER_CLASS_NAME = "com.mysql.cj.jdbc.Driver";
...

final String driverClass;
if (protocol.equals("mysql")) {
    driverClass = MYSQL_DRIVER_CLASS_NAME; // set explicitly for MySQL 8 compatibility
} else {
    driverClass = DriverManager.getDriver(unauthedUrl).getClass().getCanonicalName(); // let JDBC decide
}
```

I know that the MariaDB pages talk about the `disableMariaDbDriver` option, but that's for people who write the connection URLs to gain control without changing source code. As you're taking that control away by appending it the way you are, you might as well just explicitly select the MySQL driver, pass the connection string to that driver, and be done with it.

Reply:
Yeah, that's a slightly cleaner way to do it--I'll push that change and get rid of the options modification. 

I'm _not_ entirely opposed to leaving the part which distinguishes options from other groups in the regex, though. It leaves a proper method to interfere with flags in the future, if necessary.

Reply:
Disregarding the erroneous certificate error, this PR should be complete. The latest push includes your suggested changes and maintains the increased modularity of the new regex.